subt/odoimuloc.py

```diff
@@ -0,0 +1,84 @@
+import collections
+import logging
+import math
+import threading
+
+import osgar.lib.quaternion as quaternion
+
+g_logger = logging.getLogger(__name__)
+
+Pose2d = collections.namedtuple("Pose2d", ("x", "y", "heading"))
+
+class Localization:
+    def __init__(self, config, bus):
+        self.bus = bus
+        # outputs
+        bus.register('pose3d')
+        # inputs: origin, orientation, odom
+        self.xyz = None
+        self.orientation = None
+        self.last_odom = None
+        self.origin_xyz = None
+        self.origin_orientation = None
+        self.origin_error = False
+
+    def start(self):
+        self.thread = threading.Thread(target=self.main)
```

Comment:
do we really want Threads here? I would inherit `Localization` from Node, so that we can switch to processes instead?

Reply:
Using processes based on zmqrouter is orthogonal. The current API actually still requires that `start` starts some asynchronous processing.

Reply:
I still do not understand why do you not inherit from `Node` if it is not I/O module?

Reply:
Because I don't like `Node` :wink:. For me it is actually more work to lookup what exactly `Node` does and how to convince it to do what I want and when I manage to do that, I usually don't like how the resulting code looks. So for me the trade off is negative - it makes certain things nonobvious (-) while it makes some other things easier (+). But the (-) for me is big and the (+) is small. So for me `Node` is a half-way to somewhere but I don't actually know to where (yet?). I am thinking just having a function instead of a class might be a better abstraction but that is a discussion for a different time - something more akin to:
https://github.com/robotika/osgar/blob/677219d06c21e1985f1c4da1c354f17c84d9d259/osgar/test_zmqrouter.py#L272-L292

Reply:
Ok, I've switched to `Node`.